news/_posts/2020-06-18-about-bc-breaks.md

```diff
@@ -0,0 +1,398 @@
+
+
+### What does BC break mean
+
+If you look at issues or Pull Requests on PrestaShop GitHub repository, you might notice statements like "Sorry, we cannot do that because it would introduce a BC break" or "Cannot be done without a BC break" when discussing code changes.
+
+_You might be familiar with the concept, however if you are not, here is a short explanation:_
+
+PrestaShop is a Open Source Project that is being built with the idea that people will use it as a foundation to build their own project. Some will use it as a CMS and extend it with modules and themes. Some will use it as a shop system and customize parts of the e-commerce process. Some will even use it for either its Front-Office or Back-Office but handle the other part in another system, like an ERP or a headless CMS, and manage the communication between the two systems through API calls. So there are "external code" that uses PrestaShop "internal code" (the Core).
+
+All of these people will use code and logic that is written inside the software. For exammple, very often modules build URLs using function `Link::getModuleLink()`.
+
+However, as time passes, new versions of the software are released and this code and logic are changed to be improved, upgraded, made faster or more secure. When, from one version to another, the code changes in a way that breaks the "external code" that has been built around it, this is a Breaking Change.
+
+Here is a very simple example:
+
+```php
+public function transferPrices($product)
+```
+becoming
+```php
+public function transferPrices($product, $prices)
+```
+
+If a module uses the function `transferPrices()`, when the function changes, the module that was working previously will now crash with following error message:
+```php
+PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function transferPrices(), 1 passed and exactly 2 expected
+```
+
+So this is what a Breaking Change is: a change in the code that, when introduced into the software, might break some tools, modules, themes, whatever that is running PrestaShop Core code.
+
+So when a BC break is introduced, what happens quick is that we get lot of angry bug reports like "my code, that was working with previous version, is broken when I run it with the new version !" in GitHub!
+
+## SemVer
+
+The concept of BC Break is not specific to PrestaShop. This is actually a very common issue for all code that is being used as a foundation for other software to use: operating systems like Linux, libraries like Guzzle, tools like cURL, frameworks like Symfony ... and CMS like Drupal or PrestaShop.
+
+In order to make it easier for the people using these codebases as foundations to handle these changes, [Tom Preston Werner](http://tom.preston-werner.com/) has invented a convention: [SemVer](https://semver.org/). [SemVer](https://semver.org/) aims to help people understand immediately, when seeing a new version of software, what is contained inside it.
```

Comment:
FYI this has already been explained before https://build.prestashop.com/news/a-more-semantic-versioning-scheme/

Reply:
I think it's better to re-explain it for the reader's ease (he does not need to go read something else) but I can refer to the blog post as an external reference.
Wdyt ?
- remove this and redirect to previous blog post
- only refer to previous blog post, keep this section content